apps/draw-io/src/DrawIoEditor.vue

```diff
@@ -76,20 +89,77 @@ export default {
           this.error(error)
         })
     },
+    importVisio() {
+      const url = this.$client.files.getFileUrl(this.filePath)
+      const headers = new Headers({
+        Authorization: 'Bearer ' + this.getToken,
+        'X-Requested-With': 'XMLHttpRequest'
+      })
+      const getDescription = () =>
+        this.$gettextInterpolate(
+          'The diagram will open as a new .drawio file: %{file}',
+          { file: basename(this.filePath) },
+          true
+        )
+      // Change the working file after the import so the original file is not overwritten
+      this.filePath += `_${this.getTimestamp()}.drawio`
+      this.showMessage({
+        title: this.$gettext('Diagram imported'),
+        desc: getDescription(),
+        autoClose: {
+          enabled: true
+        }
+      })
+      fetch(url, { headers })
```

Comment:
Wouldn't this methods work here? https://github.com/owncloud/owncloud-sdk/blob/c4b96e0b50bf989e792899913fe3a70fd1698b43/src/fileManagement.js#L76

Reply:
I tried with `getFileContent`, but it looks like we cannot specify the responsetype. Drawio is able to handle the vsdx file as base64, so we need the file itself insted of its content. Also tried to re-create the file from its binary content, but I could not find a way to do so, the resultant file was always different.